flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManager.java

```diff
@@ -267,10 +267,9 @@ private void requestKubernetesPod() {
 	 * Request new pod if pending pods cannot satisfy pending slot requests.
 	 */
 	private void requestKubernetesPodIfRequired() {
-		final int requiredTaskManagerSlots = getNumberRequiredTaskManagerSlots();
-		final int pendingTaskManagerSlots = numPendingPodRequests * numSlotsPerTaskManager;
+		int requiredTaskManagers = getNumberRequiredTaskManagers();
 
-		if (requiredTaskManagerSlots > pendingTaskManagerSlots) {
+		while (requiredTaskManagers-- > numPendingPodRequests) {
```

Comment:
```suggestion
		while (requiredTaskManagers > numPendingPodRequests) {
```
I think `requestKubernetesPod` will increase `numPendingPodRequests`.